src/Compilers/Core/Portable/Collections/Rope.cs

```diff
@@ -116,6 +118,7 @@ private sealed class ConcatRope : Rope
         {
             private readonly Rope _left, _right;
             public override int Length { get; }
+            public override bool IsEmpty => _left.IsEmpty && _right.IsEmpty;
```

Comment:
>_left.IsEmpty && _right.IsEmpty [](http://example.com/codeflow?start=44&length=31)

It looks like this is going to "visit" every involved ```Rope```, but ```Length``` is already readily precomputed.